Tests/NIOCoreTests/NIOSplitLinesMessageDecoderTests.swift

```diff
@@ -0,0 +1,313 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2025 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Testing
+
+@testable import NIOCore
```

Comment:
I'd rather change `SplitMessageDecoder`'s access level to `package` than use `@testable`, but since there are `@testable` usages already in swift-nio tests, I also just used it to look more like the other tests. LMK if I should go for the `package`-access-level way.